tests/helpers/boring_model.py

```diff
@@ -151,8 +151,6 @@ def __init__(self, data_dir: str = "./"):
         self.data_dir = data_dir
         self.non_picklable = None
         self.checkpoint_state: Optional[str] = None
-
-    def prepare_data(self):
         self.random_full = RandomDataset(32, 64 * 4)
```

Comment:
This is the bugfix: accessing `datamodule.random_full` would fail on rank > 0 (see setup() below)